scripts/ci/images/Dockerfile.ci-spack-ubuntu20.04-base

```diff
@@ -1,13 +1,17 @@
 # vim: ft=dockerfile
-FROM ecpe4s/ubuntu20.04:23.05
+FROM ecpe4s/ubuntu20.04-runner-amd64-gcc-11.4:2023.08.01
+
+ARG MPICH_VERSION=4.1.1
+ARG E4S_VERSION=23.08
 
 COPY packages.yaml /etc/spack/packages.yaml
 
 # Install Base specs
-RUN . /spack/share/spack/setup-env.sh && \
+RUN git clone --depth 1 --single-branch --branch e4s-${E4S_VERSION} https://github.com/spack/spack && \
```

Comment:
spack should be already installed from the ecpe4s image, is there any reason for this clone?

Reply:
Spack is not already installed with the new image I used as a base.

Reply:
The new base image is used in spack ci as one of the runner images, and is intended to have spack cloned before each run.

Reply:
I see, any reason to use the runners image rather than the minimal spack image as we used before?

Reply:
Several reasons:

- the spack in the image we used before failed to concretize the stack with the external mpich
- according to the maintainer of the E4S images, the runner images are:
  - better: modern gnu make, patchelf, cmake, ccache, ninja exposed in default PATH
  - smaller: `1.91GB` uncompressed

Reply:
Oh that makes sense